src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -147,6 +180,29 @@ class Bounds(Generic[_T]):
     upper: _T
     """Upper bound."""
 
+    def __contains__(self, item: _T) -> bool:
+        """
+        Check if the value is within the range of the container.
+
+        Args:
+            item: The value to check.
+
+        Returns:
+            bool: True if value is within the range, otherwise False.
+        """
+        if item is None:
+            return False
+
+        assert isinstance(item, Comparable)
+
+        if self.lower is not None and self.upper is not None:
+            return self.lower <= item <= self.upper
+        if self.lower is not None:
+            return self.lower <= item
+        if self.upper is not None:
+            return item <= self.upper
+        return False
```

Comment:
I tried this without the `assert` and it fails, but I think `mypy` is wrong here, but we can help it out without having to use `isinstance()`, using [`typing.cast()`](https://docs.python.org/3.11/library/typing.html#typing.cast) instead, which doesn't need to do runtime checks (I also suggest rearranging the comparisons a bit to simplify them):
```suggestion
        if self.lower is None and self.upper is None:
            return True
        if self.lower is None:
            return item <= self.upper
        if self.upper is None:
            return self.lower <= item
        # mypy seems to get confused here, not being able to narrow upper and lower to
        # just Comparable, complaining with:
        #   error: Unsupported left operand type for <= (some union)
        # But we know if they are not None, they should be Comparable, and actually mypy
        # is being able to figure it out in the lines above, just not in this one, so it
        # should be safe to cast.
        return cast(Comparable, self.lower) <= item <= cast(Comparable, self.upper)
```